lib/utils.js

```diff
@@ -767,6 +776,36 @@ exports.stackTraceFilter = function() {
   };
 };
 
+//
+/**
+ * Number.MAX_SAFE_INTEGER only available in ES2015.
+ * @see https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
+ * @type {number}
+ */
+exports.MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || 0x1fffffffffffff;
+
+/**
+ * Attempt to convert `value` an integer, if it's a representation
+ * of a "safe" unsigned integer.
+ * @param {*} [value] Value to convert
+ * @see https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/parseInt
+ * @returns {(number|NaN)} result
+ */
+exports.toSafeUnsignedInteger = function toSafeUnsignedInteger(value) {
+  var intVal;
+  if (/^0[xX]/.test(value)) {
+    intVal = parseInt(value, 16);
+  } else if (/^0[^.]/.test(value)) {
+    intVal = parseInt(value, 8);
+  } else {
+    intVal = parseInt(value, 10);
+  }
```

Comment:
It feels like this could be simplified a bit.

```suggestion
  var base = 10;
  var intVal;
  if (/^0[xX]/.test(value)) {
    base = 16;
  } else if (/^0[^.]/.test(value)) {
    base = 8;
  }
  
  intVal = parseInt(value, base);
```